### PR TITLE
allow some percentage of failed pods so big deployments do not random…

### PR DESCRIPTION
**Pull request by grosser**
…ly fail

@zendesk/compute 

```
[19:44:59] Now deploying other roles ...
[19:44:59] Deploying GroupK role server
[19:44:59] Waiting for pods to be created
[19:44:59] Ignored 1 notready pods
[19:44:59] Deploy status:
[19:44:59]   GroupK server: Missing
[19:44:59] SUCCESS
```

### CODE REVIEW

File: plugins/kubernetes/README.md [hunk 1]
```diff
@@ -211,3 +211,8 @@ Use an annotation to configure what will to be replaced:
 metadata.annotations.samson/set_via_env_json-metadata.labels.custom: SOME_ENV_VAR
 ```
 Then configure an ENV var with that same name and a value that is valid JSON.
+
+### Ignore randoly not-ready pods during redines check
```
Comment (adammw) [inline plugins/kubernetes/README.md]: typo

File: plugins/kubernetes/app/models/kubernetes/deploy_executor.rb [hunk 2]
```diff
@@ -88,7 +88,8 @@ def wait_for_resources_to_complete(release_docs, timeout)
           @output.puts "No pods were created"
           return success, statuses
         end
-        not_ready = statuses.reject(&:live)
+
+        not_ready = not_ready_by_percent(statuses)
```
[the remark this answers is not in the document]
Comment (grosser) [reply plugins/kubernetes/app/models/kubernetes/deploy_executor.rb]: tried that at first and looked complicated ... ended up refactoring it and now this seems like the nicer solution ... will change

**Comment by grosser** [comment]
updated

File: plugins/kubernetes/test/models/kubernetes/deploy_executor_test.rb [hunk 3]
```diff
@@ -675,6 +675,26 @@ def worker_is_unstable
...
   end
 
+  describe "#allowed_not_ready" do
+    let(:log_string) { "Ignored" }
+
+    it "ignores none when percent is not set" do
+      executor.send(:allowed_not_ready, 10).must_equal 0
+    end
+
+    it "ignores given percentage" do
```
Comment (adammw) [inline plugins/kubernetes/test/models/kubernetes/deploy_executor_test.rb]: Ignores sounds wrong in this context, should be like returns number based on ..

**Review by adammw — APPROVED** [review]

**Pull request merged**